doc/tutorials/highlevel-bindings.md

```diff
@@ -54,6 +54,34 @@ sufficient to write (or generate) a one-to-one mirror of the C API and use that
 The generated code should still be understandable. So if writing a more idiomatic API on top of the
 direct mapping, significantly simplifies the generated code (and maybe also the template), you should write such an API.
 
+### Bindings for the `lowlevel` Library
+
+Your programming language of choice must provide a way to call into C code (like [cgo](https://golang.org/cmd/cgo/)).
+
+In general we prefer (in this order):
+1. Automatically and statically generated bindings (like [swig](http://www.swig.org/)).
+1. Manually written bindings that directly call to library without any statically build part (like JNI and GI). They are a bit slower but can directly access libelektra.so.
+1. Manually written bindings that are built statically.
+
+If you want to manually write a binding make sure you have a good understanding of the possible limitations the interop layer can have (e.g. variadic functions).
+
+What you will also need is to set up the Compiler + Linker flags, for this I can recommend [pkg-config](https://www.freedesktop.org/wiki/Software/pkg-config/), because Elektra already provides `.pc` files.
+
+For garbage collected (GC) languages freeing memory by hand is not something you usually do and since the GC has no knowledge of memory allocated in C we have two options:
+* Forcing the user to call the appropriate functions like `keyDel()` themselves, which is very developer unfriendly and error prone or
+* Using language featurse like Java / C++ Destructors or Go's `runtime.SetFinalizer()` function to automatically and reliably release the memory as soon as the native objects are garbage collected.
+
+If you decide on mapping the functionality of kdb.h 1:1 it is pretty straight-forward - whereas if you want to adapt or enhance some API's to leverage language features like iterators or operator overloading feel free to do so. The less 'alien' the binding feels to its users the better.
+
+Remember that Elektra has internal iterators (for metadata+keysets) but in general we prefer external iterators by either copying the KeySet per iterator or using `ksAtCursor`.
+
+### Variadic functions
```

Comment:
```suggestion
### Variadic Functions
```

Did you format this already with cmake-format?

Reply:
you mean `reformat-markdown`? Did it now

Reply:
Yes, you can also run `reformat-all` to not miss any reformatting tool.